frontend/src/components/VideoChat.tsx

```diff
@@ -0,0 +1,362 @@
+    });
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    const copyThis = this;
+
+    const peerNode = this.peerVideoRef.current;
+    peer.on("stream", function(stream) {
+      //when we get stream from other user, create new video
+      copyThis.setState({ peerConnected: true });
+      console.log("copythis.peerconnected: " + copyThis.state.peerConnected);
+      // // createVideo(stream)
+      // const peerNode = this.peerVideoRef.current;
+      if (peerNode) {
+        peerNode.srcObject = stream;
+      }
+    });
+    peer.on("close", () => {
+      //when peer is closed, destroy video
+      alert("PEER CLOSED");
+      copyThis.setState({ peerConnected: false });
+      copyThis.setState({ inVideoChat: false });
+      console.log("copythis.peerconnected: " + copyThis.state.peerConnected);
+      // socket.emit('close');
+      peer.destroy(); //Everything is cleaned up,
+    });
+
+    peer.on("error", err => {
+      // alert(err);
+      copyThis.setState({ inVideoChat: false });
+    });
+    return peer; //Return our peer
+  };
+
+  //for peer of type init
+  makePeer = (): void => {
+    const gotAnswer = this.state.gotAnswer;
+    const socket = this.socket;
+    const videoChatId = this.state.videoChatId;
+    const peer = this.init(PeerTypes.init);
+    peer.on("signal", function(data) {
+      if (!gotAnswer) {
+        //TODO: somehow include id in this emit offer
+        const offerObj = { data: data, videoChatId: videoChatId };
+        socket.emit("Offer", offerObj);
+      }
+    });
+    this.setState({ peer: peer });
+  };
+
+  frontAnswer = (offer: any): void => {
+    const peer = this.init(PeerTypes.notInit);
+    const socket = this.socket;
+    const videoChatId = this.state.videoChatId;
+    //this doesnt run automatically, have to call signal
+    peer.on("signal", data => {
+      //means we have gotten our offer
+      //we want to imitate with even Answer
+      //TODO: Include id in this emit
+      const answerObj = { data: data, videoChatId: videoChatId };
+      socket.emit("Answer", answerObj);
+      // socket.emit('Answer', data);
+    });
+    //pass offer to signal, generate answer to backend, which will send to other user
+    peer.signal(offer);
+  };
+
+  // handles when answer comes from backend
+  // if answer comes from backend, set the client.gotAnswer to true, and signalAnswerView to peer,
+  // then finally both the clients will be connected
+  signalAnswer = (answer: any): void => {
+    this.setState({ gotAnswer: true });
+    const peer = this.state.peer;
+    peer?.signal(answer);
+  };
+
+  createVideo = (stream: any): void => {
+    const peerNode = this.peerVideoRef.current;
+    if (peerNode) {
+      peerNode.srcObject = stream;
+      this.setState({ peerVideo: stream });
+    }
+  };
+
+  render() {
+    const { classes, users } = this.props;
+    return (
+      <React.Fragment>
+        <h1>Video Chat</h1>
+        {!this.state.inVideoChat && (
```

Comment:
Could replace this conditional and the one on line 301 with a ternary operator

Reply:
Lazy... I think ill leave this so i can get on to styling/small features/bug fixes for videochat